src/benchmarklib/tpch/tpch_table_generator.cpp

```diff
@@ -274,4 +274,23 @@ std::unordered_map<std::string, BenchmarkTableInfo> TPCHTableGenerator::generate
   return table_info_by_name;
 }
 
+AbstractTableGenerator::IndexesByTable TPCHTableGenerator::_indexes_by_table() const {
+  return {
+      {"part", {{"p_partkey"}}},
+      {"supplier", {{"s_suppkey"}, {"s_nationkey"}}},
+      // TODO(anyone): multi-column indexes are currently not used by the index scan rule and the translator
+      {"partsupp", {{"ps_partkey", "ps_suppkey"}, {"ps_suppkey"}}},  // ps_partkey is subset of {ps_partkey, ps_suppkey}
+      {"customer", {{"c_custkey"}, {"c_nationkey"}}},
+      {"orders", {{"o_orderkey"}, {"o_custkey"}}},
+      {"lineitem", {{"l_orderkey", "l_linenumber"}, {"l_partkey", "l_suppkey"}}},
+      {"nation", {{"n_nationkey"}, {"n_regionkey"}}},
+      {"region", {{"r_regionkey"}}},
+  };
+}
+
+AbstractTableGenerator::SortOrderByTable TPCHTableGenerator::_sort_order_by_table() const {
+  // Allowed as per TPC-H Specification, paragraph 1.5.2
```

Comment:
Partitioning on date columns is (1.5.4) and partitioning without pruning seems pointless.

Reply:
But with partitioning that’s logical pruning based on the part criterion. I guess Jan asks about auxiliary data structures.
Any idea if they are allowed? I’d guess no.

Reply:
Can't find a definite yes or no. However, statistics must be allowed (otherwise join ordering wouldn't work). I can't find anything that keeps us from using those statistics for other purposes. The TPC-H Analyzed Paper talks explicitly about filters, too.

Reply:
In my opinion, [CP3.2](https://homepages.cwi.nl/~boncz/snb-challenge/chokepoints-tpctc.pdf) also strongly suggests that this is legal. Also, how is this different from pruning a chunk if the value is not in the dictionary?